build.rs

```diff
@@ -195,13 +196,25 @@ fn get_roc_app_object(target: &RocRaySupportedTarget) -> AppType {
             }
         }
         RocRaySupportedTarget::Linux => {
-            // Check if the app.o built by roc exists
+            // Check which app built by roc exists
             let app_o = manifest_dir().join("app.o");
-            let app_so = manifest_dir().join("libapp.so");
+            let lib_app_o = manifest_dir().join("libapp.o");
+
             if app_o.exists() {
                 AppType::Static(app_o.to_path_buf())
-            } else if app_so.exists() {
-                AppType::Dynamic(app_so.to_path_buf())
+            } else if lib_app_o.exists() {
+                let cwd_libapp_so = manifest_dir().join("libapp.so");
+
+                let output = std::process::Command::new("ld")
+                    .args(&["-shared", "-o", &cwd_libapp_so.to_string_lossy(), "libapp.o"])
+                    .output()
+                    .expect("Failed to execute linker command");
+
+                assert!(output.status.success(), "{output:#?}");
+                assert!(output.stdout.is_empty(), "{output:#?}");
+                assert!(output.stderr.is_empty(), "{output:#?}");
+
+                AppType::Dynamic(cwd_libapp_so)
```

Comment:
So the dylib produced when we use `roc build --lib` isn't working? we have to make a shared library using `ld` instead?

Reply:
I was just wrong about this one.